clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -653,11 +654,23 @@ void PointerVariableConstraint::setTypedef(TypedefNameDecl* T, std::string s) {
 // variables and potentially nested function pointer declaration. Produces a
 // string that can be replaced in the source code.
 
+std::string PointerVariableConstraint::gatherQualStrings(void) const {
```

Comment:
This looks suspicious. It looks like it squashes all of the qualifier for the ConstraintVariable from potentially different pointer levels into the same string. Is it correct when there are qualifiers on a bunch of different pointer levels?

Reply:
I see what you're saying but I can't come up with an example it doesn't work for?

Reply:
With this latest change the function here is unnecessary, you can just inline what you have. I think there's a version that doesn't require an ostream, I've used `.getType().getQualifiers().getAsString()` before.